### PR TITLE
bump the version of the k8s go client and generated code

### PR DESCRIPTION
**Pull request by steeling**
Bump the version In prep for https://github.com/openservicemesh/osm/pull/4673

### CODE REVIEW

**Review by shashankram — APPROVED** [review]

**Comment by shashankram** [comment]
Needs to be rebased.

**Review by nojnhuh — COMMENTED** [review]
I can think of one more spot to update: https://github.com/openservicemesh/osm/blob/2cfae97c71d58ff7199f80a08279563c26e146fb/dockerfiles/Dockerfile.osm-crds#L5

**Comment by steeling** [comment]
> https://dl.k8s.io/release/v1.22.2/bin

done!

File: dockerfiles/Dockerfile.osm-crds [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM --platform=$BUILDPLATFORM busybox:1.33 AS builder
 ARG TARGETPLATFORM
 # Talking to the internet in an arm64 container doesn't seem to work from a
 # amd64 Mac, so download the kubectl binary in a stage running the native arch.
-RUN wget https://dl.k8s.io/release/v1.22.2/bin/$TARGETPLATFORM/kubectl -O /bin/kubectl && \
+RUN wget https://dl.k8s.io/release/v1.23.5/bin/linux/amd64/kubectl -O /bin/kubectl && \
```
Comment (nojnhuh) [inline dockerfiles/Dockerfile.osm-crds:5]: This still needs `$TARGETPLATFORM` so arm64 builds work.

Comment (steeling) [reply dockerfiles/Dockerfile.osm-crds]: oops, bad copy/pasta

Comment (steeling) [reply dockerfiles/Dockerfile.osm-crds]: fixed now

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**